src/functions.php

```diff
@@ -301,10 +326,20 @@ function fun($filter, $parameters = null)
  */
 function remove($filter)
 {
-    if (@\stream_filter_remove($filter) === false) {
+    $errstr = '';
+    \set_error_handler(function ($_, $error) use (&$errstr) {
+        // Match errstr from PHP's warning message.
+        // stream_filter_remove() expects parameter 1 to be resource,...
+        $errstr = $error;
+    });
+
+    $ret = \stream_filter_remove($filter);
+
+    \restore_error_handler();
+
+    if ($ret === false) {
         // PHP 8 throws above on type errors, older PHP and memory issues can throw here
```

Comment:
This suggest PHP 8 type errors aren't currently taken care of as the `stream_filter_remove()` would throw, can you look into this?

Reply:
Done 👍